server/mediaserver.go

```diff
@@ -136,7 +140,37 @@ func (s *LivepeerServer) StartMediaServer(ctx context.Context, maxPricePerSegmen
 //RTMP Publish Handlers
 func createRTMPStreamIDHandler(s *LivepeerServer) func(url *url.URL) (strmID string) {
 	return func(url *url.URL) (strmID string) {
-		id, err := core.MakeStreamID(core.RandomVideoID(), "RTMP")
+		//Create a StreamID
+		//If streamID is passed in, use that one
+		//Else create one
+		var vid []byte
+		qid := core.StreamID(url.Query().Get("hlsStrmID"))
+		if qid != "" {
+			vid = qid.GetVideoID()
+			if vid == nil {
+				glog.Error("Error parsing hlsStrmID")
+				return ""
+			}
+		} else {
+			vid = core.RandomVideoID()
+		}
+
+		// Ensure there's no concurrent StreamID with the same name
+		mid, err := core.MakeManifestID(vid)
+		if err != nil {
+			glog.Error("Error constructing manifest ID", err)
+			return ""
+		}
+		s.connectionLock.Lock()
```

Comment:
Doesn't look like we're writing here - should this be `s.connectionLock.RLock()`?

Reply:
Yup, fixed